File: solutions/csharp/space-age/1/SpaceAge.cs
// Dedicated to Shree DR.MDD — the light behind every shadow

public record SpaceAge(ulong ChronoSpan)
{
    private const ulong OrbitalBase = 31557600;

    public double OnEarth() => RevolvedByRatio(1.0d);

    public double OnMercury() => RevolvedByRatio(0.2408467d);

    public double OnVenus() => RevolvedByRatio(0.61519726d);

    public double OnMars() => RevolvedByRatio(1.8808158d);

    public double OnJupiter() => RevolvedByRatio(11.862615d);

    public double OnSaturn() => RevolvedByRatio(29.447498d);

    public double OnUranus() => RevolvedByRatio(84.016846d);

    public double OnNeptune() => RevolvedByRatio(164.79132d);

    private double RevolvedByRatio(double orbitalFactor) => ChronoSpan / orbitalFactor / OrbitalBase;
}

File: solutions/java/space-age/1/src/main/java/SpaceAge.java
// For my Shree DR.MDD

import java.util.*;

class SpaceAge {
    private final Map<String, Double> ORBITAL_TIMES = Map.of(
        "Mercury", 0.2408467,
        "Venus", 0.61519726,
        "Earth", 1.0,
        "Mars", 1.8808158,
        "Jupiter", 11.862615,
        "Saturn", 29.447498,
        "Uranus", 84.016846,
        "Neptune", 164.79132
    );
    private final double totalSeconds, totalYears;

    SpaceAge(double totalSeconds) {
        this.totalSeconds = totalSeconds;
        this.totalYears = totalSeconds / 31_557_600;
    }

    double getSeconds() {
        return totalSeconds;
    }

    private double calculateYears(String planet) {
        return totalYears / ORBITAL_TIMES.get(planet);
    }

    double onEarth() {
        return calculateYears("Earth");
    }

    double onMercury() {
        return calculateYears("Mercury");
    }

    double onVenus() {
        return calculateYears("Venus");
    }

    double onMars() {
        return calculateYears("Mars");
    }

    double onJupiter() {
        return calculateYears("Jupiter");
    }

    double onSaturn() {
        return calculateYears("Saturn");
    }

    double onUranus() {
        return calculateYears("Uranus");
    }

    double onNeptune() {
        return calculateYears("Neptune");
    }
}
